PARSEC Page added to DevHub

david-maidsafe opened the pull request:
Added a new 'Stand-Alone Technology' section to the DevHub front page and a new page to host PARSEC information.

nadiaburborough commented on src/contents/en-gb/home.yaml:
```diff
@@ -46,6 +46,18 @@ contribute:
       href: /core_technology
+sharing:
+  title:
+    shared: Stand alone
```
The alignment of 'Stand alone technology' title is too low (at full width screen on laptop

david-maidsafe replied:
adjusted (was the same diff as section above but looked bigger as section is smaller)

nadiaburborough commented on src/contents/en-gb/home.yaml:
```diff
@@ -46,6 +46,18 @@ contribute:
+    desc: Use standalone components from the SAFE Network in your own projects.
+  parsec:
+    title: Protocol for Asynchronous, Reliable, Secure and Efficient Consensus
+    desc: PARSEC is an algorithm which allows a section of network nodes to reach consensus on the validity and order of network events by voting on events and using a gossip protocol to disseminate these votes amongst themselves.
```
This description goes right to the right hand edge of the screen on a laptop screen at full size.

S-Coyle replied:
Reproduced this cross browser - it's also easy to resize and end up with the description overhanging the edge of the screen, when every other section of the page has resized appropriately. Or open the page in iPad Pro portrait dimensions - 1024x1366.

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+pageIntro:
+  title: What is PARSEC?
+  desc: PARSEC is a Protocol for Asynchronous, Reliable, Secure and Efficient Consensus and is the world's first open source ABFT protocol that can be used in a truly permissionless setting created by MaidSafe.
```
font here doesn't match the rest of the page

david-maidsafe replied:
it's the page header - I'll address (btw - just notes for me as I'm going through these)

david-maidsafe replied:
format changed to match

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+networkUse:
+  title: Why does the Network need it?
+  para1: As a global decentralised data and communications network, the SAFE Network has to be public and permissionless.
+  para2: Unlike centralised systems there is no central body to act as a final authority on the state of the Network.  Nodes is a decentralised system such as the SAFE Network must therefore agree on Network actions and state amongst themselves. Other decentralised systems such as BitCoin use blockchain to achieve consensus.  However, there is a number of drawbacks with a blockchain based system  ...
```
Needs a space above this para

nadiaburborough replied:
Nodes ~is~ `on` a decentralised

nadiaburborough replied:
therefore agree on ~N~ `n`etwork actions

nadiaburborough replied:
However, there ~is~ `are` a

nadiaburborough replied:
cost as ~blockchain~ `it` grows

david-maidsafe replied:
all but leading space done

david-maidsafe replied:
format changed to match

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+  para1: As a global decentralised data and communications network, the SAFE Network has to be public and permissionless.
+  para2: Unlike centralised systems there is no central body to act as a final authority on the state of the Network.  Nodes is a decentralised system such as the SAFE Network must therefore agree on Network actions and state amongst themselves. Other decentralised systems such as BitCoin use blockchain to achieve consensus.  However, there is a number of drawbacks with a blockchain based system  ...
+bigDeal:
+  title: Why is it a big deal?
```
Why is ~it~ `PARSEC` a big deal?

david-maidsafe replied:
done

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+  para2: Unlike centralised systems there is no central body to act as a final authority on the state of the Network.  Nodes is a decentralised system such as the SAFE Network must therefore agree on Network actions and state amongst themselves. Other decentralised systems such as BitCoin use blockchain to achieve consensus.  However, there is a number of drawbacks with a blockchain based system  ...
+bigDeal:
+  title: Why is it a big deal?
+  para1: As far as we know no other project has managed to ensure ABFT consensus in permissionless environments.  This is a technology that will power the SAFE Network, the only truly next-generation decentralised autonomous internet that promises privacy, security and freedom by default.
```
environments. This is ~a~ technology that will power the SAFE Network ~,~ `:`

david-maidsafe replied:
done

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+  title: Why is it a big deal?
+  para1: As far as we know no other project has managed to ensure ABFT consensus in permissionless environments.  This is a technology that will power the SAFE Network, the only truly next-generation decentralised autonomous internet that promises privacy, security and freedom by default.
+myProject:
+  title: Great, can I use it on my project?
```
can I use it ~on~ `in` my project?

david-maidsafe replied:
done

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+resources:
+  title:
+    dark: PARSEC
+    light1: PARSEC Resources
```
different format to the other titles

S-Coyle commented on src/constants.js:
```diff
@@ -20,6 +20,14 @@ export default {
+      resources: {
+        github: 'https://github.com/maidsafe/parsec',
+        rfc: 'https://github.com/maidsafe/rfcs/blob/master/text/0049-parsec/0049-parsec.md',
+        whitepaper: 'http://docs.maidsafe.net/Whitepapers/pdf/PARSEC.pdf',
```
Whitepaper URL should be updated to https

S-Coyle replied:
Confirmed resolved

S-Coyle commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,66 @@
+      logo: GITHUB
+      linkId: github
+    - name: RFC
+      logo: RFC
```
Tested the new /parsec page on osx Safari and noticed that the RFC, Whitepaper & Documentation images at the bottom of the page all flicker a lot when I scroll. The GitHub image is fine.
Not able to reproduce this on Chrome or FF, but confirmed a colleague could also repro on Safari

S-Coyle replied:
images have been removed from the page now - problem no longer visible.

S-Coyle commented on src/contents/en-gb/core_technology.yaml:
```diff
@@ -34,6 +34,12 @@ components:
+      chunk1: In order to come to this eventual consistency the routing layer uses the consensus mechanism
+      link1:
+        name: PARSEC
+        href: '/parsec'
```
This opens the /parsec page in a new browser tab - just confirming whether this is intentional?

david-maidsafe replied:
wasn't just copied and pasted the link line from above but it doesn't make sense to open a new tab. Changed.

david-maidsafe commented:
@S-Coyle & @nadiaburborough - I have addressed the points in your change requests. If you could confirm the changes and (on acceptance) would Stephen be able to do the merge?

nadiaburborough commented on src/contents/en-gb/home.yaml:
```diff
@@ -46,6 +46,18 @@ contribute:
+    desc: Use standalone components from the SAFE Network in your own projects.
+  parsec:
+    title: PARSEC
+    desc: PARSEC (Protocol for Asynchronous, Reliable, Secure and Efficient Consensus) is an algorithm which allows a section of network nodes to reach consensus on the validity and order of network events by voting on events and using a gossip protocol to disseminate these votes amongst themselves.
```
Sorry to be picky .. but this sentence needs some commas and/or full stops to help the reader digest the info. Or provide just a summary sentence here (e.g. PARSEC (Protocol for Asynchronous, Reliable, Secure and Efficient Consensus) is an algorithm which allows nodes to reach consensus in a decentalised network.), and provide the fuller explanation on the page.

david-maidsafe replied:
Na, it's fair comment.

david-maidsafe replied:
done

nadiaburborough commented on src/contents/en-gb/parsec.yaml:
```diff
@@ -0,0 +1,63 @@
+  title: PARSEC
+pageIntro:
+  title: What is PARSEC?
+  desc: PARSEC is a Protocol for Asynchronous, Reliable, Secure and Efficient Consensus and is the world's first open source ABFT protocol that can be used in a truly permissionless setting created by MaidSafe.
```
Suggest link from `ABFT protocol` to what that means

david-maidsafe replied:
Expanded out the initialism

lionel1704 commented:
@david-maidsafe from a technical perspective, the changes made are good to go!

david-maidsafe commented:
@S-Coyle - made changes as requested and commit submitted.
Ready for Merge I guess - @shankar2105?
